independent-projects/arc/runtime/src/main/java/io/quarkus/arc/MapBeanSupplier.java

```diff
@@ -8,17 +8,18 @@
  *
  * @author Maarten Mulders
  */
-public class MapBeanSupplier implements Supplier {
-    private final Map map;
+public class MapBeanSupplier implements Supplier<Object> {
```

Comment:
Why not make it a `Supplier<Supplier<InjectableBean<?>>`, so the type parameter corresponds with the `beanIdToBeanSupplier` map?

Reply:
Good point. It has no importance but I can change it... ;-)

Reply:
In fact, we could change it to `Map<String,T>` so that it could be theoretically reused.

Reply:
Even better :-)